src/fu-util-common.c

```diff
@@ -393,6 +393,63 @@ fu_util_update_reboot (GError **error)
 	return val != NULL;
 }
 
+gboolean
+fu_util_prompt_confirm (FwupdDevice *device, FwupdRelease *release, GError **error)
+{
+	const gchar *desc_tmp;
+	gint vercmp;
+	g_autoptr(GString) str = g_string_new (NULL);
+
+	vercmp = fu_common_vercmp_full (fwupd_release_get_version (release),
+				        fu_device_get_version (device),
+				        fwupd_device_get_version_format (device));
+	if (vercmp < 0) {
+		/* TRANSLATORS: message letting the user know an downgrade is available
+		 * %1 is the device name and %2 and %3 are version strings */
+		g_string_append_printf (str, _("Downgrade %s from %s to %s:"),
+					fwupd_device_get_name (device),
+					fwupd_device_get_version (device),
+					fwupd_release_get_version (release));
+	} else if (vercmp > 0) {
+		/* TRANSLATORS: message letting the user know an upgrade is available
+		 * %1 is the device name and %2 and %3 are version strings */
+		g_string_append_printf (str, _("Upgrade %s from %s to %s:"),
+					fwupd_device_get_name (device),
+					fwupd_device_get_version (device),
+					fwupd_release_get_version (release));
+	} else {
+		/* TRANSLATORS: message letting the user know an upgrade is available
+		 * %1 is the device name and %2 is a version string */
+		g_string_append_printf (str, _("Reinstall %s to %s:"),
+					fwupd_device_get_name (device),
+					fwupd_release_get_version (release));
+	}
+
+	/* no description provided */
```

Comment:
I don't follow this comment, what does it really mean?

Reply:
It's supposed to be `description is optional` and the logic was wonky, fixed.